src/mainClass.js

```diff
@@ -83,21 +77,11 @@ export class AllItems {
   displayLinks(category) {
     getData(category).then((res) => {
       this.displayChosenLinks(res);
-      this.chosenLinks = res;
-      this.onReceivedChosenLinks();
-    })
-  }
```

Comment:
Ok this solved to we have single method for receiving links with param category. This is ok, now try to

this.onReceivedLinks - why need to accept category?

try to implement filter prop now, and getLinks which returns all or filtered dependend of filter prop.

And that result is used on ui.

So changing search param changes filter prop, and calls getLinksn again for update ui.
Click on clear set filter to null/empty and calls getLinks again which is full list